src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackHandlerFactory.java

```diff
@@ -38,7 +38,8 @@ public CallbackResponse dispatch(CallbackParams callbackParams) {
 
     private boolean hasSupportedRoles(CallbackHandler callbackHandler, String authorisation, String eventId) {
         List<String> userRoles = userService.getUserDetails(authorisation).getRoles();
-        if (callbackHandler.getSupportedRoles().stream().anyMatch(userRoles::contains)) {
+        if (callbackHandler.getSupportedRoles().stream().anyMatch(role ->
+            userRoles.contains(role.getRole()))) {
             return true;
         } else {
             throw new ForbiddenActionException("User does not have supported role for event " + eventId);
```

Comment:
Looks like this method's responsibility is just to return `false` here and let the invoker decide what to do with it. Otherwise, this could follow a pattern we've used elsewhere like `void verifyHasSupportedRoles(...) throws X`